__tests__/data/ga4.json

```diff
@@ -10962,5 +10958,307 @@
+            "name": "RudderLabs JavaScript SDK",
+            "version": "1.0.0"
+          },
+          "locale": "en-US",
+          "os": {
+            "name": "iOS",
+            "version": "14.4.1"
+          },
+          "screen": {
+            "density": 2
+          },
+          "externalId": [
+            {
+              "type": "ga4AppInstanceId",
+              "id": "f0dd99b6f979fb551ce583373900f937"
+            },
+            {
+              "type": "ga4ClientId",
+              "id": "client_id"
+            }
+          ],
+          "userAgent": "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_2) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/79.0.3945.88 Safari/537.36"
+        },
+        "type": "track",
+        "event": "product added",
+        "properties": {
+          "currency": "USD",
+          "price": 2.4,
+          "quantity": 2,
+          "products": [
+            {
+              "product_id": "507f1f77bcf86cd799439011",
+              "name": "Monopoly: 3rd Edition",
+              "coupon": "SUMMER_FUN",
+              "category": "Apparel",
+              "brand": "Google",
+              "variant": "green",
+              "price": "19",
+              "quantity": "2",
+              "position": "1",
+              "affiliation": "Google Merchandise Store",
+              "currency": "USD",
+              "discount": 2.22,
+              "item_category2": "Adult",
+              "item_category3": "Shirts",
+              "item_category4": "Crew",
+              "item_category5": "Short sleeve",
+              "item_list_id": "related_products",
+              "item_list_name": "Related Products",
+              "location_id": "L_12345"
+            }
+          ]
+        },
+        "integrations": {
+          "All": true
+        }
+      },
+      "destination": {
+        "Config": {
+          "apiSecret": "QyWKGHj8QhG2L4ePAPiXCA",
+          "measurementId": "G-T40PE6BET4",
+          "firebaseAppId": "",
+          "blockPageViewEvent": false,
+          "typesOfClient": "gtag",
+          "extendPageViewParams": false,
+          "sendUserId": false,
+          "eventFilteringOption": "disable",
+          "blacklistedEvents": [
+            {
+              "eventName": ""
+            }
+          ],
+          "whitelistedEvents": [
+            {
+              "eventName": ""
+            }
+          ],
+          "enableServerSideIdentify": true,
+          "sendLoginSignup": true,
+          "newOrExistingUserTrait": "firstLogin",
+          "loginSignupMethod": "method",
+          "generateLead": true,
+          "generateLeadValueTrait": "value",
+          "generateLeadCurrencyTrait": "currency",
+          "clientIdFieldIdentifier": "properties.client_id"
+        },
+        "Enabled": true
+      }
+    },
+    "output": {
+      "statusCode": 400,
+      "error": "[Google Analytics 4]: ga4ClientId, anonymousId, context.device.id or messageId must be provided",
```

Comment:
This test case is not correct as we need to take `client_id` from fallback values. Here, we have `"anonymousId": "ea5cfab2-3961-4d8a-8187-3d1858c99090",` defined in the input payload

Reply:
updated